compiler/natives/src/syscall/syscall_js_wasm.go

```diff
@@ -0,0 +1,77 @@
+//go:build js
+
+package syscall
+
+import (
+	"syscall/js"
+)
+
+func runtime_envs() []string {
+	process := js.Global().Get("process")
+	if process.IsUndefined() {
+		return nil
```

Comment:
One way to look at this is that "no `process`" is functionally equivalent to "no environment variables set", which could happen in a regular OS environment (although it rarely does). Things I'd expect to break when no environment variables are set are mostly filesystem-related like `$HOME`. There's no file system in the browser either, so they'd break anyway ¯\_(ツ)_/¯

I don't think there is a good alternative to environment variables in the browser context, which we could use as a shim. However, I believe `os.Getenv()` and `os.Setenv()` should work as expected in a browser, so at least the user code would be able to read variables it sets itself.

Prohibiting access to environment variables in the browser doesn't seem feasible because there are several places in the runtime that attempt to read them and handle empty values correctly.

It's not the most satisfying answer to me, but I think this is one of the cases where GopherJS users have to be aware of the runtime environment they are targeting and take extra steps to handle it correctly.

Reply:
> I believe os.Getenv() and os.Setenv() should work as expected in a browser, so at least the user code would be able to read variables it sets itself.

I agree with this.

I think my main concern would be if someone is depending on a third-party library that does something silly like setting an env variable, then reading it back, and getting an empty string instead of what should be there. If we handle the above case, this would be taken care of.